### PR TITLE
RHMI service endpoints alerts

### PR DESCRIPTION
**Pull request by obrienrobert**
## Additional Information
[INTLY-9909](https://issues.redhat.com/browse/INTLY-9909)

Creation of RHMI service endpoints alerts.

### Install
- Install using this branch
- Ensure that none of the newly created alerts are firing during install
- Ensure all of the required alerts are created. [This](https://docs.google.com/spreadsheets/d/1Wt6qrRskOQLd5DndMXvMvocsfsxcIOk7uk8TWsX4MBo/edit#gid=2054611867) Google Sheet (1.x tab) contains all of the endpoint alerts
- To test each alerts individually, you can either:

    1. Go to the specific product namespace `-> resources -> search for endpoints -> actions -> Edit YAML`. In the YAML editor, delete everything from `subsets` downwards. This will trigger the related alert after 5 minutes
    2. Scale the products deployment to 0 to trigger the alert, and then back up to the desired number of pods to clear the alert. This 

- Ensure the the SOP URL for each alert is also correct [(669)](https://github.com/RHCloudServices/integreatly-help/pull/669)

### Upgrade
- Install the latest release on your cluster
- Upgrade to this branch
- Verify using point 3-5 from the install verification above

### Installation Verification
- Ensure the author of the PR has attached a log of the installation run from his branch to the jira or pr and check that it exited as expected.
- Verify the fresh installation is correct on cluster provided by PR author 
### Upgrade Verification
- After installation verification, notify the PR author to begin an upgrade on their cluster
- Ensure the developer of the PR has attached a log of the upgrade run from his branch to the jira or pr and check that it exited as expected. 
- If possible, look at the tasks that ran and see they match the PR
- Verify the upgrade is correct on cluster provided by PR author 

## Checklist

- [x] Updated [Changelog](https://github.com/integr8ly/installation/blob/master/CHANGELOG.md)
- [x] Tested Installation
- [x] Tested Uninstallation
- [ ] Tested or Created follow on for Upgrade

Is an upgrade task required and are there additional steps needed to test this?
- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by jhellar** [comment]
The SOP urls point to `...sops/service_endpoint_down.asciidoc`, but will be `...sops/alerts/service_endpoint_down.asciidoc`, once https://github.com/RHCloudServices/integreatly-help/pull/669/files is merged.

**Comment by jhellar** [comment]
Verified installation part so far. What I haven't tested yet was `no alerts during install` and alert for prometheus service endpoint.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ Some of these changes may include:
 ## Unreleased
 
 ### Added
+* [INTLY-9909] - Creation of RHMI service endpoints alerts
 
 ### Changed
 * [INTLY-3623] - Refactor of inventories and associated group_vars to support POC, OSD and PDS environments
```
Comment (b1zzu) [inline CHANGELOG.md:19]: @obrienrobert should we reset the changelog for the 1.8.1?

Comment (obrienrobert) [reply CHANGELOG.md:19]: @b1zzu do you mean I should remove the changelog entry in this PR?

Comment (b1zzu) [reply CHANGELOG.md:19]: I actually don't know, but it feels like all the changelog that has been released should be removed

@ciaran-byrne do you know how the changelog works in 1.x?

Comment (b1zzu) [reply CHANGELOG.md:19]: looking at the history of the changelog it seems like nothing has ever been removed, but then I don't get why the `Unreleased` section

**Comment by jhellar** [comment]
For the verification of alerts firing during installation, these are results I got:

- installation took ~1h
- no alerts firing in first ~50min
- during last ~10min there were various alerts firing

Log:

```
Po srp 31 11:46:18 UTC 2020


Po srp 31 11:47:19 UTC 2020


Po srp 31 11:48:19 UTC 2020


Po srp 31 11:49:20 UTC 2020


Po srp 31 11:50:20 UTC 2020


Po srp 31 11:51:21 UTC 2020


Po srp 31 11:52:21 UTC 2020


Po srp 31 11:53:21 UTC 2020


Po srp 31 11:54:22 UTC 2020


Po srp 31 11:55:22 UTC 2020


Po srp 31 11:56:23 UTC 2020


Po srp 31 11:57:23 UTC 2020


Po srp 31 11:58:24 UTC 2020


Po srp 31 11:59:24 UTC 2020


Po srp 31 12:00:25 UTC 2020


Po srp 31 12:01:25 UTC 2020


Po srp 31 12:02:26 UTC 2020


Po srp 31 12:03:26 UTC 2020


Po srp 31 12:04:27 UTC 2020


Po srp 31 12:05:27 UTC 2020


Po srp 31 12:06:28 UTC 2020


Po srp 31 12:07:28 UTC 2020


Po srp 31 12:08:28 UTC 2020


Po srp 31 12:09:29 UTC 2020


Po srp 31 12:10:29 UTC 2020


Po srp 31 12:11:30 UTC 2020


Po srp 31 12:12:30 UTC 2020


Po srp 31 12:13:31 UTC 2020


Po srp 31 12:14:31 UTC 2020


Po srp 31 12:15:32 UTC 2020


Po srp 31 12:16:32 UTC 2020


Po srp 31 12:17:33 UTC 2020


Po srp 31 12:18:33 UTC 2020


Po srp 31 12:19:34 UTC 2020


Po srp 31 12:20:34 UTC 2020


Po srp 31 12:21:35 UTC 2020


Po srp 31 12:22:35 UTC 2020


Po srp 31 12:23:36 UTC 2020


Po srp 31 12:24:36 UTC 2020


Po srp 31 12:25:36 UTC 2020


Po srp 31 12:26:37 UTC 2020


Po srp 31 12:27:37 UTC 2020


Po srp 31 12:28:38 UTC 2020


Po srp 31 12:29:38 UTC 2020


Po srp 31 12:30:39 UTC 2020


Po srp 31 12:31:39 UTC 2020


Po srp 31 12:32:40 UTC 2020


Po srp 31 12:33:40 UTC 2020


Po srp 31 12:34:41 UTC 2020


Po srp 31 12:35:41 UTC 2020


Po srp 31 12:36:42 UTC 2020


Po srp 31 12:37:42 UTC 2020
"RHMIThreeScaleBackendRedisServiceEndpointDown" "RHMIThreeScaleSystemMySqlServiceEndpointDown" "RHMIThreeScaleZyncDatabaseServiceEndpointDown" "RHMIManagedServiceBrokerServiceEndpointDown"

Po srp 31 12:38:43 UTC 2020
"RHMIThreeScaleBackendRedisServiceEndpointDown" "RHMIThreeScaleSystemMySqlServiceEndpointDown" "RHMIThreeScaleZyncDatabaseServiceEndpointDown"

Po srp 31 12:39:43 UTC 2020


Po srp 31 12:40:44 UTC 2020


Po srp 31 12:41:46 UTC 2020
"RHMIAmqOnlinePostgreSqlServiceEndpointDown"

Po srp 31 12:42:46 UTC 2020
"RHMIAmqOnlinePostgreSqlServiceEndpointDown"

Po srp 31 12:43:47 UTC 2020
"RHMISsoServiceEndpointDown" "RHMISsoPostgreSqlServiceEndpointDown"

Po srp 31 12:44:48 UTC 2020
"RHMISsoServiceEndpointDown" "RHMISsoPostgreSqlServiceEndpointDown"

Po srp 31 12:45:49 UTC 2020
"RHMIUserSsoServiceEndpointDown" "RHMIUserSsoPostgresSqlServiceEndpointDown"

Po srp 31 12:46:49 UTC 2020
"RHMIUserSsoServiceEndpointDown" "RHMIUserSsoPostgresSqlServiceEndpointDown"

Po srp 31 12:47:50 UTC 2020
"RHMIUserSsoServiceEndpointDown"

Po srp 31 12:48:50 UTC 2020
"RHMIUserSsoServiceEndpointDown"

Po srp 31 12:49:51 UTC 2020


Po srp 31 12:50:51 UTC 2020


```

**Comment by obrienrobert** [comment]
@jhellar thanks for the review. Would it possible that the alerts were in a `pending` state during install? I think `pending` alerts are also classified as `firing`, which might explain why the shell command returned so many alerts.

File: CHANGELOG.md (hunk 1 above)
Comment (obrienrobert) [reply CHANGELOG.md:19]: @ciaran-byrne just following up on this to see if the changelog is OK as is in this PR, or if changes are required?

Comment (ciaran-byrne) [reply CHANGELOG.md:19]: If there is a new/updated SOP it would be helpful to link it the Changelog

Comment (obrienrobert) [reply CHANGELOG.md:19]: I've updated the change-log entry to also link to the accompanying SOP for these new `endpoint` alerts.

**Comment by obrienrobert** [comment]
@david-martin would it be possible for you to approve these changes whenever you get a chance, please?

**Pull request merged**